exercise-integration-testing-5-53/test/spread-the-code-for-task-10/when-event-is-valid.test.js

```diff
@@ -0,0 +1,270 @@
+// const request = require("supertest");
+const axios = require("axios");
+const { initializeAPI } = require("../../sensors-api");
+const nock = require("nock");
+
+let App;
+let port;
+let server;
+
+beforeAll(async () => {
+  const { expressApp, thePort, theServer } = await initializeAPI();
+  App = expressApp;
+  port = thePort;
+  server = theServer;
+});
+
+afterAll((done) => {
+  server.close(() => {
+    done();
+  });
+});
+
+beforeEach(() => {
+    nock("http://localhost").get("/notification").reply(200, {
+      success: true,
+    });
+  });
+  
+  function helper(length) {
+    var randomChars =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    var result = "";
+    for (var i = 0; i < length; i++) {
+      result += randomChars.charAt(
+        Math.floor(Math.random() * randomChars.length)
+      );
+    }
+    return result;
+  }
+  describe("Sensors test", () => {
+    // ✅ TASK: Test that when a new event is posted to /event route, if category or temperature are not specified -> the API returns HTTP 400
+    // 💡 TIP: Down below, there is an example event schema
+    
+  
+    // ✅ TASK: Test that when a new event is posted to /sensor-events route, the temperature is not specified -> the event is NOT saved to the DB!
+    // 💡 TIP: Testing the response is not enough, the adequate state (e.g. DB) should also satisfy the expectation
+  
+    
+  
+    // ✅ TASK: Test that when a new valid event is posted to /sensor-events route, we get back a valid response
+    // 💡 TIP: Consider both the HTTP status and the body
+  
+    test("When new valid event is send,  should get back ", async () => {
+      //Arrange
+      const eventToAdd = {
+        category: "kids-room",
+        temperature: 20,
+        name: "Thermostat-temperature", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+      //Act
+  
+      const config = {
+        validateStatus: () => true,
+      };
+      const url = `http://localhost:${port}/sensor-events`;
+      const res = await axios.post(url, eventToAdd, config);
+  
+      //Assert
+      expect(res.status).toBe(200);
```

Comment:
Not checking the body? This is where a good test title helps, when clarifying to yourself what do you want to test. One test case can cover both the status and the body (toMatchObject({status:..., body:...})